core/arch/arm/kernel/boot.c

```diff
@@ -1252,6 +1252,10 @@ void init_tee_runtime(void)
 	if (!IS_ENABLED(CFG_VIRTUALIZATION))
 		call_preinitcalls();
 	call_initcalls();
+
+	/* crypto_rng_read() should be ready to be used by now */
+	thread_init_core_local_pauth_keys();
```

Comment:
the functions are stubbed when config switches are disabled.

Reply:
sure but just paraphrassing @jenswi-linaro review to one of my PR
"I prefer a configuration where the overhead can be reduced to zero when not needed. With this weak function always being called, we're still adding a few useless instructions for every platform. If this call was guarded with some IS_ENABLED() test, like before, it could be optimized away completely if not needed."

Reply:
more over since the PR is disabling the feature by default..
https://github.com/OP-TEE/optee_os/pull/5548/commits/1fd4e7de4c3858a41ee3afbcc2da04f68be39a36#diff-6301a2703b709daa4d84ead1c30bf67cdaf088abed68753a8f9c740f1e58b54aR871

Reply:
> sure but just paraphrassing @jenswi-linaro review to one of my PR "I prefer a configuration where the overhead can be reduced to zero when not needed. With this weak function always being called, we're still adding a few useless instructions for every platform. If this call was guarded with some IS_ENABLED() test, like before, it could be optimized away completely if not needed."
> 
> why should this be any different?

It's different because it's done differently. I'm stubbing using a static inline function, that has zero overhead with optimizations enabled.

Reply:
> should it be surrounded with an `if (IS_ENABLED(CFG_CORE_PAUTH)))` to make it easier to read? I realize it is functionally equivalent...

That can be discussed, I chose to just call the function in this case. I thought it would be pretty clear what's going on in this case.